vspace/src/main/java/edu/asu/diging/vspace/core/aspects/ExhibitionDataAspect.java

```diff
@@ -49,37 +55,47 @@ public void setExhibition(JoinPoint jp) {
         Class<?> returnType = signature.getReturnType();
         // only if there is a model object injected and if we're returning a string
         // (assuming that returning a string implies rendering of a view afterwards
-        if(args!=null && returnType==String.class) {
-            for(Object obj : args) {
-                if(obj instanceof Model && !(obj instanceof RedirectAttributes)) {
-                    if(!((Model) obj).containsAttribute("exhibition")) {
+        if (args != null && returnType == String.class) {
+            for (Object obj : args) {
+                if (obj instanceof Model && !(obj instanceof RedirectAttributes)) {
+                    if (!((Model) obj).containsAttribute("exhibition")) {
                         ((Model) obj).addAttribute("exhibition", exhibitionManager.getStartExhibition());
                     }
                     if (!((Model) obj).containsAttribute("publishedSpaces")) {
-                        List<ISpace> publishedSpaces=spaceManager.getSpacesWithStatus(SpaceStatus.PUBLISHED);
-                        /* (non-Javadoc)
-                         * Added to show spaces with null status and accommodate existing spaces with null space status
+                        List<ISpace> publishedSpaces = spaceManager.getSpacesWithStatus(SpaceStatus.PUBLISHED);
+                        /*
+                         * (non-Javadoc) Added to show spaces with null status and accommodate existing
+                         * spaces with null space status
                          */
                         publishedSpaces.addAll(spaceManager.getSpacesWithStatus(null));
                         ((Model) obj).addAttribute("publishedSpaces", publishedSpaces);
                     }
                 }
-            }  
+            }
         }
     }
 
     @Around("execution(public * edu.asu.diging.vspace.web.publicview..*Controller.*(..))")
     public Object showExhibition(ProceedingJoinPoint jp) throws Throwable {
+
+        HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes())
+                .getRequest();
+        Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition();
+        Map pathVariables = (Map)request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+        String previewId= (String) pathVariables.get("previewId");  
+        if (request.getRequestURI().contains("/preview/") && (exhibition.getPreviewId() == null || !exhibition.getPreviewId().equals(previewId))) {
+            return jp.proceed();
+        }
```

Comment:
actually, I think all of the above code can be simplified. It looks like you are always injecting the previewId as variable into the controller, so you can just check if that is set and if so, show the preview.